waku-org/nwaku
# fix: filter - enhancements in subscription management

## Description

Railgun notified about filter issues. This PR tackled that.
Railgun informed that after 4 days working properly, the nodes stopped sending messages. There were also issues that caused a max of 100 messages being sent over one connection and the reason for that was that we were `dialPeer`ing on every single message the filter server sent to the filter subscribers.

## Changes

* waku_filter_v2: idiomatic way run periodic subscription manager
* filter subscriptions: add more debug logs
* filter: make sure the custom start and stop procs are called
* make sure filter protocol is started if it is mounted
* filter: dial push connection onsubscribe only
* waku/waku_filter_v2/client.nim: make the client to behave as a server for the FilterPush messages sent by filter server

## How to test

1. Start a nwaku with wss enabled
2. [lab.waku.org](https://github.com/waku-org/lab.waku.org) -> examples/experimental/railgun-repro -> npm run dev
3. Press F12 and you should see messages comming without interruption

## Issue

closes https://github.com/waku-org/nwaku/issues/3191
closes https://github.com/waku-org/nwaku/issues/3192
